### ISSUE TITLE
Push to PyPI workflow

### ISSUE THREAD

**Issue by domire8** [aica-technology/api#57]
After #56, the workflow failed: https://github.com/aica-technology/api/actions/runs/6551381044/job/17792438097

It doesn't seem that this workflow was updated yet to only run if the version of the api client has changed. Could you have a look at that? @jjenscodee

**Comment by eeberhard** [comment]
Agreed, it would be preferable to skip the push-to-pypi workflow if the python project version has not changed.

**Comment by jjenscodee** [comment]
So we will need to parse and check the `pyproject.toml`, correct?

**Comment by domire8** [comment]
Yeah I think the `pyproject.toml` is where the version is stored.

I don't think that this will be enough, we need to check on push to main if the version in the toml file has changed and if so, create a new tag and push to pypi.
